cfme/tests/automate/test_git_import.py

```diff
@@ -78,3 +91,70 @@ def test_automate_git_domain_displayed_in_service(appliance, imported_domain, br
     view.modal.apply.click()
     assert view.provisioning_entry_point.value == ("/{}/Service/Provisioning/StateMachines/"
         "ServiceProvision_Template/CatalogItemInitialization".format(imported_domain.name))
+
+
+@pytest.mark.tier(1)
+def test_refresh_git_current_user(imported_domain, new_user):
+    """
+    Polarion:
+        assignee: ghubale
+        initialEstimate: 1/8h
+        caseimportance: medium
+        caseposneg: positive
+        testtype: functional
+        startsin: 5.9
+        casecomponent: Automate
+        tags: automate
+        testSteps:
+            1. created non-super user 'user1' along with default 'admin' user.
+            2. Using admin user imported git repo.:
+               'https://github.com/ramrexx/CloudForms_Essentials.git' or any other repo.
+            3. Logged in with admin and refreshed domain- 'CloudForms_Essentials' or other domain.
+               Then checked all tasks.
+            4. Found user name 'admin' next to 'Refresh git repository'.
+            5. Then checked instances in that domain by logging in with user 'user1' and 'admin'.
+            6. Logged in with non-super user 'user1' and refreshed domain - 'CloudForms_Essentials'.
+               Then checked all tasks.
+            7. Found user name 'user1' next to 'Refresh git repository'.
+            8. Then checked instances in that domain by logging in with user 'user1' and 'admin'.
+        expectedResults:
+            1.
+            2.
+            3.
+            4.
+            5. It shows that
+               e.g. 'Automate Instance [Provisioning - Updated 2019-01-15 11:41:43 UTC by admin]'
+            6.
+            7.
+            8. It shows that
+               e.g. 'Automate Instance [Provisioning - Updated 2019-01-15 11:44:43 UTC by user1]'
+               Hence, correct user that calls refresh automation domain from git branch is shown.
+    Bugzilla:
+        1592428
+    """
+    tasks_collection = imported_domain.appliance.collections.tasks
+
+    # Refreshed imported domain by 'Admin' user
+    imported_domain.refresh(branch_or_tag='Branch', git_branch='origin/master')
+    view = imported_domain.create_view(DomainDetailsView)
+    view.flash.assert_message("Successfully refreshed!")
+    view = navigate_to(tasks_collection, 'AllTasks')
+
+    # Collecting list of all tasks performed by users
+    all_tasks = view.tabs.alltasks.table.read()
+    for task in all_tasks:
```

Comment:
Q:
what if expected `task` not found in `all_tasks`. test will still pass. do you not think we need proper assertion?